flex-checks/src/main/java/org/sonar/flex/checks/FileHeaderCheck.java

```diff
@@ -67,23 +76,48 @@ public void setCharset(Charset charset) {
 
   @Override
   public void init() {
-    expectedLines = headerFormat.split("(?:\r)?\n|\r");
+    if (isRegularExpression) {
+      regularExpression = Pattern.compile(headerFormat);
+    } else {
+      expectedLines = headerFormat.split("(?:\r)?\n|\r");
+    }
   }
 
   @Override
   public void visitFile(AstNode astNode) {
-    List<String> lines = Collections.emptyList();
+    File file = getContext().getFile();
+    boolean hasHeader = isRegularExpression ? matchesRegularExpression(file) : matchesPlainTextHeader(file);
+    if (!hasHeader) {
+      getContext().createFileViolation(this, "Add or update the header of this file.");
+    }
+  }
 
+  @VisibleForTesting
+  protected boolean matchesRegularExpression(File file) {
+    String content;
     try {
-      lines = Files.readLines(getContext().getFile(), charset);
+      content = Files.toString(file, charset);
     } catch (IOException e) {
-      LOG.error("Unable to execute rule \"S1451\" for file {} because of error: {}",
-        getContext().getFile().getName(), e);
+      throw new IllegalStateException(exceptionMessage(file), e);
+    }
+    Matcher matcher = regularExpression.matcher(content);
```

Comment:
@lindamartin we just check that the file starts with an expected text: it can be code or copyright or any text, that's the current behavior of that rule (before we add anything related to regular expressions).